### ISSUE TITLE
Memory leak detected by ASAN not detected by CiFuzz 

### ISSUE THREAD

**Issue by bimbashrestha** [google/oss-fuzz#3813]
We added a new fuzzer to zstd and it had a memory leak that CIFuzz didn't catch. We were able to find it with local ASAN. CIFuzz instead reported some other error. See the "Run Fuzzers" error in our pipeline: https://github.com/facebook/zstd/runs/676119909

You can reproduce this by checking out https://github.com/facebook/zstd/pull/2140/commits/f35a1f189e6e6f3c7f245f350aa42e43a34a6174

Running our local fuzzer (https://github.com/facebook/zstd/tree/dev/tests/fuzz) with --enable-asan catches the leak but CIFuzz does not.

Linking where we add the new fuzzer: https://github.com/facebook/zstd/pull/2140

**Comment by evverx** [comment]
Judging by "TESTCASE=/github/workspace/out/leak-278fe0c22af16a2fb46831fd11e1662693357ca0" it seems CIFuzz managed to find the memory leak. Though it looks like CIFuzz downloads and unpacks the latest zstd build from GCS from https://storage.googleapis.com/clusterfuzz-builds/zstd/zstd-address-202005140759.zip and then expects to find `dictionary_stream_round_trip` (which isn't there because it's a new fuzz target). `is_reproducible` should probably return `False` if `os.path.join(target_path, self.target_name)` doesn't exist.

**Comment by jonathanmetzman** [comment]
Thank you @bimbashrestha for reporting this with important details and thank you @evverx for the helpful analysis. Working on a fix now.

**Comment by evverx** [comment]
As an aside, I wonder if it's safe to assume that contributors adding new fuzz targets are generally willing to fix bugs (or wait for someone else to fix them) before their PRs can be merged. It would probably make sense to somehow allow projects to choose whether to ignore bugs found by new fuzz targets or not (at least until all "shallow" bugs have been reported on Monorail).

**Comment by jonathanmetzman** [comment]
@evverx put your comment in a new issue so I won't forget about it.
My fixes is delayed by a bit of refeactoring I am doing on the tests.
@bimbashrestha I don't think you are blocked by this issue right?

**Comment by bimbashrestha** [comment]
> @bimbashrestha I don't think you are blocked by this issue right?

No, not a blocker. Thanks @jonathanmetzman

**Comment by jonathanmetzman** [comment]
OK, i'll land this on monday thne.

**Comment by evverx** [comment]
@jonathanmetzman I've just come across https://github.com/systemd/systemd/pull/15226/checks?check_run_id=709118544 where a new fuzz target is being added. Could you land the fix?

**Comment by jonathanmetzman** [comment]
There's a bit of cleanup I need to do on the patch to make it landable. I'll try to do this today.
